[be] feat: 카테고리 목록 조회

yhpark95 opened the pull request:
#26

yhpark95 committed:
feat: 카테고리 목록 조회 기능 구현

#26

yhpark95 committed:
test: 카테고리 목록 조회 테스트 구현

#26

yeonise commented on kiosk-be/src/main/java/team04/kioskbe/Drink/DrinkController.java:
```diff
@@ -0,0 +1,21 @@
+        this.drinkService = drinkService;
+    }
+
+    @GetMapping("/Drinks/Categories")
+    public List<CategoryResponse> getCategories() {
+        return drinkService.getCategories();
+    }
```
url 소문자~~~용

AlbertImKr commented on kiosk-be/src/main/java/team04/kioskbe/Drink/CategoryResponse.java:
```diff
@@ -0,0 +1,19 @@
+package team04.kioskbe.Drink;
+
+public class CategoryResponse {
+    private String id;
```
setter이 없으면 final을 선언하면 좋을 것 같아용.

AlbertImKr commented on kiosk-be/src/main/java/team04/kioskbe/Drink/DrinkService.java:
```diff
@@ -0,0 +1,14 @@
+public class DrinkService {
+
+    public List<CategoryResponse> getCategories(){
+        return Category.getCategoryList();
```
getCategoryList() -> getCategories() 로 변경하면 좋을 것 같아용.

list는 -> Clollection을 지정하여 유연하지 않다고 판단해용.

AlbertImKr commented on kiosk-be/src/main/java/team04/kioskbe/domain/Category.java:
```diff
@@ -23,4 +28,12 @@ public String getId() {
         return name;
     }
+
+    public static List<CategoryResponse> getCategoryList(){
+        List<CategoryResponse> list = new ArrayList<>();
+        for(Category category : Category.values()){
+            list.add(new CategoryResponse(category.id, category.name));
+        }
+        return list;
```
Stream을 사용해서 반환하면 더 예쁠 것 같아용.

AlbertImKr replied:
CategoryResponse DTO에서 category 도메인을 참조하는 것이 좋을 것 같아용.

AlbertImKr commented on kiosk-be/src/test/java/team04/kioskbe/Drink/DrinkServiceTest.java:
```diff
@@ -0,0 +1,33 @@
+    @DisplayName("카테고리 목록 조회")
+    public void getCategoriesTest() {
+        //given
+        List<CategoryResponse> givenList = new ArrayList<>();
+        givenList.add(new CategoryResponse("coffee", "커피"));
+        givenList.add(new CategoryResponse("latte", "라떼"));
+        givenList.add(new CategoryResponse("tea", "티"));
+        givenList.add(new CategoryResponse("juice", "쥬스"));
+        givenList.add(new CategoryResponse("decaffeine", "디카페인"));
```
결과이여서 given에 있는 것 보다는 then에 있는것이 맞을 것 같아용??

yhpark95 committed:
refactor: 카테고리 목록 조회 리뷰 반영

#26
- URL 소문자로 변경
- final 추가
- 메소드 네이밍 변경
- for문 스트림으로 변경
- 카테고리 조회 메소드 CategoryResponse로 이동
- 테스트 코드 수정
